projects/core/src/user/services/find-product-page-meta.resolver.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { Observable, of, combineLatest } from 'rxjs';
+import { filter, map, switchMap } from 'rxjs/operators';
+import {
+  PageMetaResolver,
+  PageTitleResolver,
+  PageBreadcrumbResolver,
+} from '../../cms';
+import { PageMeta, Page } from '../../cms/model/page.model';
+import { TranslationService } from '../../i18n/translation.service';
+import { PageType } from '../../model/cms.model';
+import { RoutingService } from '../../routing';
+import { ProductSearchService } from '../../product/facade/product-search.service';
+import { AuthService } from '../../auth/facade/auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FindProductPageMetaResolver extends PageMetaResolver
+  implements PageTitleResolver, PageBreadcrumbResolver {
+  totalAndCode$: Observable<[number, any]> = combineLatest([
+    this.productSearchService.getResults().pipe(
+      filter(data => !!(data && data.pagination)),
+      map(results => results.pagination.totalResults)
+    ),
+    this.routingService.getRouterState().pipe(
+      map(state => state.state.queryParams['couponcode']),
+      filter(Boolean)
+    ),
+  ]);
+
+  constructor(
+    protected routingService: RoutingService,
+    protected productSearchService: ProductSearchService,
+    protected translation: TranslationService,
+    protected authService: AuthService
+  ) {
+    super();
+    this.pageType = PageType.CONTENT_PAGE;
+    this.pageTemplate = 'SearchResultsListPageTemplate';
+  }
+
+  /**
+   * @deprecated since version 1.3
```

Comment:
the resolve() comes from its parent class PageMetaResolver which deprecated it since 1.3. So we use 1.3 tag here. in fact it's useless here and can be deleted safely when it's deleted from the parent class.